utils/wasm-gen/src/config/syscalls/injection.rs

```diff
@@ -0,0 +1,116 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Entities describing possible injection types for each sys-call.
+//!
+//! Types here are used to create [`crate::SysCallsConfig`].
+
+use crate::InvocableSysCall;
+
+use gear_wasm_instrument::syscalls::SysCallName;
+use std::{collections::HashMap, ops::RangeInclusive};
+
+/// Possible injection type.
+#[derive(Debug, Clone)]
+pub enum InjectionType {
```

Comment:
`InjectionType` type name actually represents idea properly, but it's unclear if you read the crate for the first time. What about `SysCallInjectionKind`?

Reply:
Can I keep it as `SysCallInjectionType`? "Kind" will probably lead to renaming of `SysCallsInjectionTypes` 
done in 1217f73872b7ef3adeeec2f8f4f2a966a10b6a92

Reply:
kinds weren't bad, actually.